### PR TITLE
Detect null pointer exceptions in #7422

### PR DESCRIPTION
**Pull request by t0yv0**
# Description

Fixes # (issue)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by t0yv0** [comment]
CC @gitfool

**Comment by t0yv0** [comment]
This runs counter to expectations.. Our own test suite fails on this with:

```
xUnit.net 00:01:06.39]     Pulumi.Automation.Tests.LocalWorkspaceTests.InlineProgramAllowsParallelExecution [FAIL]
  X Pulumi.Automation.Tests.LocalWorkspaceTests.InlineProgramAllowsParallelExecution [3s 194ms]
  Error Message:
   System.AggregateException : One or more errors occurred. (Attempting to repeatedly set `Urn` property on Pulumi.Stack resource. Please report to https://github.com/pulumi/pulumi/issues)
---- System.Exception : Attempting to repeatedly set `Urn` property on Pulumi.Stack resource. Please report to https://github.com/pulumi/pulumi/issues
  Stack Trace:
```

File: sdk/dotnet/Pulumi/Resources/Resource.cs [hunk 1]
```diff
@@ -20,46 +20,70 @@ public class Resource
...
         [Output(Constants.UrnPropertyName)]
-        public Output<string> Urn { get; private protected set; } = null!;
+        public Output<string> Urn {
+            get
+            {
+                if (_urn != null)
+                {
+                    return _urn;
+                }
+                throw new Exception($"Uninitialized `Urn` property on {this} resource. " +
```
Comment (gitfool) [inline sdk/dotnet/Pulumi/Resources/Resource.cs:72]: This should be wrapped in an `else` block.

File: sdk/dotnet/Pulumi/Resources/Resource.cs [hunk 2]
```diff
@@ -20,46 +20,70 @@ public class Resource
...
+                                    "Please report to https://github.com/pulumi/pulumi/issues");
+            }
+            private protected set
+            {
+                if (_urn == null)
+                {
+                    _urn = value;
+                }
+                throw new Exception($"Attempting to repeatedly set `Urn` property on {this} resource. " +
```
Comment (gitfool) [inline sdk/dotnet/Pulumi/Resources/Resource.cs:81]: This should be wrapped in an `else` block.

Comment (t0yv0) [reply sdk/dotnet/Pulumi/Resources/Resource.cs:81]: Ah certainly, thanks :/ here it's not just style is it.

Comment (gitfool) [reply sdk/dotnet/Pulumi/Resources/Resource.cs]: The first else on the getter wasn't needed... I'm still waking up! 😉

Comment (t0yv0) [reply sdk/dotnet/Pulumi/Resources/Resource.cs]: Fixed up, let's see if the CI passes possibly.

**Pull request closed without merging**